src/ros_license_toolkit/main.py

```diff
@@ -112,9 +113,14 @@ def main(args: Optional[Sequence[str]] = None) -> int:
     rll_print(f'Execution time: {stop - start:.2f} seconds', Verbosity.QUIET)
 
     # Print the overall results
-    if all(results_per_package.values()):
+    if all(result == Status.SUCCESS
+           for result in results_per_package.values()):
         rll_print(f"All packages:\n {SUCCESS_STR}", Verbosity.QUIET)
         return os.EX_OK
+    if all(result != Status.FAILURE
+           for result in results_per_package.values()):
+        rll_print(f"All packages:\n {WARNING_STR}", Verbosity.QUIET)
+        return os.EX_OK
     rll_print(f"All packages:\n {FAILURE_STR}", Verbosity.QUIET)
     return os.EX_DATAERR
```

Comment:
What would this do with warnings right now?
Please also use the `max(results_per_package.values())` approach here. And then make 3 different outputs (also one for warning)

Reply:
It is splitted into strings for success, warning and failure. Shall i add something to the warning string? How else do you want to differ the outputs?